drivers/ethernet/phy/phy_tja1103.c

```diff
@@ -411,17 +382,19 @@ static int phy_tja1103_init(const struct device *dev)
 
 static int phy_tja1103_link_cb_set(const struct device *dev, phy_callback_t cb, void *user_data)
 {
+	int rc = 0;
 	struct phy_tja1103_data *const data = dev->data;
 
 	data->cb = cb;
 	data->cb_data = user_data;
 
-	/* Invoke the callback to notify the caller of the current
-	 * link status.
-	 */
-	invoke_link_cb(dev);
+	rc = phy_tja1103_get_link_state(dev, &data->state);
 
-	return 0;
+	if (rc == 0) {
+		data->cb(dev, &data->state, data->cb_data);
+	}
```

Comment:
might also here be better to just use the state of the data struct and not fetch it and leave that to the phy_work_handler